spec/duration.html

```diff
@@ -52,5 +87,182 @@ <h1>Duration Objects</h1>
         <li>milliseconds</li>
         <li>microseconds</li>
         <li>nanoseconds</li>
-    </ul>
+    </ul> -->
+  <emu-clause id="sec-temporal-duration-constructor">
+    <h1>The Temporal.Duration Constructor</h1>
+    <p>
+      The Temporal.Duration constructor is the %Temporal.Duration% intrinsic
+      object. When called as a constructor, it creates and initializes a new
+      Temporal.Duration object.
+    </p>
+    <p>
+      The Temporal.Duration constructor is designed to be subclassable. It may
+      be used as the value of an extends clause of a class definition. Subclass
+      constructors that intend to inherit the specified Duration behaviour must
+      include a super call to the %Temporal.Duration% constructor to create and
+      initialize subclass instances with the necessary internal slots.
+    </p>
+    <emu-clause id="sec-temporal-duration-constructor-alg">
+      <h1>
+        Temporal.Duration (_years_, _months_, _days_, _hours_, _minutes_,
+        _seconds_, _milliseconds_, _microseconds_, _nanoseconds_)
+      </h1>
+      <p>
+        When the Temporal.Duration function is called, the following steps are
+        taken:
+      </p>
+      <emu-alg>
+        1. If NewTarget is *undefined*, then
+          1. Throw a *TypeError* exception.
+        1. Let _h_ be ? ToInteger(_hours_).
+        1. Let _m_ be ? ToInteger(_minutes_).
+        1. Let _s_ be ? ToInteger(_seconds_).
+        1. Let _ms_ be ? ToInteger(_milliseconds_).
+        1. Let _mis_ be ? ToInteger(_microseconds_).
+        1. Let _ns_ be ? ToInteger(_nanoseconds_).
+        1. Let _bt_ be ? BalanceTime(_h_, _m_, _s_, _ms_, _mis_, _ns_).
+        1. Let _y_ be ? ToInteger(_years_).
+        1. Let _mo_ be ? ToInteger(_months_).
+        1. Let _d_ be ? ToInteger(_days_).
```

Comment:
Actually, another thought. I think we should do all the ToInteger calls before the BalanceTime call, and do them in argument order

Reply:
Okay.